modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -634,25 +657,70 @@ public BatchResponsePart doExecuteChangeSet(final BatchHandler handler, final Li
         }
     }
 
+    private boolean isErroneousResponse(ODataResponse response) {
+        // erroneous requests are >= 400 (BAD_REQUEST)
+        return response != null && response.getStatus().getStatusCode() >= HttpStatusCodes.BAD_REQUEST.getStatusCode();
+    }
+
+    private void checkForKeys(ODataEntry entry, List<EdmProperty> keyProperties) throws ODataException {
+        if (!keyProperties.isEmpty()) {
+            for (EdmProperty keyProperty : keyProperties) {
+                if (entry.getProperties().get(keyProperty.getName()) == null) {
+                    throw new ODataException("Cannot create entity without key(s)");

```

Comment:
add info about the entry like name or id to the exception message, it will be easier to debug it.